### DISCUSSION TITLE
Howto address Buttons on V1.2 ESP32 Lora Modul Ipex GPS-NEO-M8N SX1278 433MHz and devel20240316-B17-full.bin?

### DISCUSSION THREAD

**Discussion by georgjorge** [dl9rdz/rdz_ttgo_sonde#437] [TTGO firmware]
The board has three buttons: RST, GPIO 38 and Power. No matter how I set Button input and Button 2 input the only reaction I get is via Button Power to scan from one display to the next. I am not able to get the reaction described under the Control Menu. That means there is no way to get the frequency screen by medium keypress.
Any idea?

**Comment by dl9rdz** [comment]
This should work but has never been tested.
The V1.2 (assume this is aT-Beam Board) uses a different PMU chip, and the power button is handled via the PMU chip.
I will have a look later this week.

**Comment by georgjorge** [comment]
Yes, it is a T-Beam Board V1.2 ESP32 Lora Modul.

**Comment by OK1FWG** [comment]
I compiled the devel20240419 and on my TTGO-T-Beam T22 V1.1 and when set the screen saver I am not able to wake up the display using any button, the screen remain blank. Left comment on the topic in Issues.

**Reply by dl9rdz** [reply]
What time out do you use? It is working fine for me, but you have to use a screen saver time out larger than 1 second (otherwise it will remain blank, as the display is updated only once per second, so if you have something like "1" or "11" in the config, this is 0 or 1 seconds, so it will not work).

**Reply by dl9rdz** [reply]
Maybe to explain better: config value:
1 => display is always off
2 => display is always off if no signal is received
and then e.g.
51 => display goes off after 5 seconds

**Comment by dl9rdz** [comment]
The buttons seem to work as intended also on the T-Beam V1.2
(https://github.com/dl9rdz/rdz_ttgo_sonde/wiki/Interactive-control)

In the default configuration, the power button (the one of the left) is used as button 2 - which means that only a short keypress has an assigned function, the medium keypress is detected correctly (but in the standard screens config no action is associated with button 2 medium keypress).
The middle button - button 1 - works as described in the interactive control, allowing you to switch to WiFi scan, spectrum display, next frequency, and back to frequency scan.
